removeOldTempIndexes on PageStore causes NullPointerException

Occures on: Version: 1.4.197
Startup Parameters: `MV_STORE=FALSE;MVCC=FALSE;AUTO_RECONNECT=TRUE;DB_CLOSE_DELAY=-1;TRACE_LEVEL_FILE=0`
## Issue
On the startup, the `PageStore` tries to clean up old temporary indexes from transactions, which just where about to happen on the last run, when e.g. a power fail of the system occurred. This although causes a `NullPointerException`.

## Stacktrace
```
Caused by: java.lang.NullPointerException
        at org.h2.engine.Database.unlockMeta(Database.java:947)
        at org.h2.engine.Session.commit(Session.java:704)
        at org.h2.store.PageStore.removeOldTempIndexes(PageStore.java:402)
        at org.h2.store.PageStore.openExisting(PageStore.java:374)
        at org.h2.store.PageStore.open(PageStore.java:289)
        at org.h2.engine.Database.getPageStore(Database.java:2552)
        at org.h2.engine.Database.open(Database.java:709)
        at org.h2.engine.Database.openDatabase(Database.java:286)
        ... 49 more
```

## Explanation
On [`Database.open()L709`](https://github.com/h2database/h2database/blob/version-1.4.197/h2/src/main/org/h2/engine/Database.java#L709) , the `PageStore` db gets opened during which the mentioned clean up of temporary indexes happens on [`PageStore.openExisting()L374`](https://github.com/h2database/h2database/blob/version-1.4.197/h2/src/main/org/h2/store/PageStore.java#L374). In case of existing temporary indexes, a `Session.commit()` gets executed after they've been removed. As rows got removed the table gets marked for analyzation during commit, which finally causes the call [`Database.unlockMeta()947`](https://github.com/h2database/h2database/blob/version-1.4.197/h2/src/main/org/h2/engine/Database.java#L947). This causes a `NullPointerException` because `meta` is still null at that point as it only gets initialized after [`Database.open()L709`](https://github.com/h2database/h2database/blob/version-1.4.197/h2/src/main/org/h2/engine/Database.java#L709) , specifically on [`Database.open()L758`](https://github.com/h2database/h2database/blob/version-1.4.197/h2/src/main/org/h2/engine/Database.java#L758)

```
ERROR: new TestCrashAPI().init(test).testCase(1567052917); // Bug -1339964437 id=-1657 callCount=1623 openCount=15 General error: "java.lang.NullPointerException" [50000-197] org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.NullPointerException" [50000-197] ------------------------------
org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.NullPointerException" [50000-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.engine.Database.openDatabase(Database.java:345)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(DriverManager.java:571)
	at java.sql.DriverManager.getConnection(DriverManager.java:215)
	at org.h2.test.synth.TestCrashAPI.getConnection(TestCrashAPI.java:230)
	at org.h2.test.synth.TestCrashAPI.testCase(TestCrashAPI.java:309)
	at org.h2.test.synth.TestCrashAPI.test(TestCrashAPI.java:176)
	at org.h2.test.TestBase.runTest(TestBase.java:139)
	at org.h2.test.TestAll.addTest(TestAll.java:1028)
	at org.h2.test.TestAll.test(TestAll.java:856)
	at org.h2.test.TestAll.runTests(TestAll.java:655)
	at org.h2.test.TestAll.testAll(TestAll.java:563)
	at org.h2.test.TestAll.run(TestAll.java:513)
	at org.h2.test.TestAll.main(TestAll.java:454)
Caused by: java.lang.NullPointerException
	at org.h2.engine.Database.unlockMeta(Database.java:1086)
	at org.h2.engine.Session.analyzeTables(Session.java:729)
	at org.h2.engine.Session.commit(Session.java:717)
	at org.h2.store.PageStore.removeOldTempIndexes(PageStore.java:395)
	at org.h2.store.PageStore.openExisting(PageStore.java:367)
	at org.h2.store.PageStore.open(PageStore.java:288)
	at org.h2.engine.Database.getPageStore(Database.java:2741)
	at org.h2.engine.Database.open(Database.java:768)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	... 22 more
```

--- a/h2/src/main/org/h2/engine/Database.java
+++ b/h2/src/main/org/h2/engine/Database.java
@@ -1082,9 +1082,11 @@ public class Database implements DataHandler {
      * @param session the session
      */
     public void unlockMeta(Session session) {
-        unlockMetaDebug(session);
-        meta.unlock(session);
-        session.unlock(meta);
+        if (meta != null) {
+            unlockMetaDebug(session);
+            meta.unlock(session);
+            session.unlock(meta);
+        }
     }
 
     /**
@@ -1888,7 +1890,7 @@ public class Database implements DataHandler {
                 }
             }
         } else {
-            lockMeta(session);
+            boolean metaWasLocked = lockMeta(session);
             synchronized (this) {
                 int id = obj.getId();
                 removeMeta(session, id);
@@ -1897,6 +1899,9 @@ public class Database implements DataHandler {
                 if(id > 0) {
                     objectIds.set(id);
                 }
+            }
+            if (!metaWasLocked) {
+                unlockMeta(session);
             }
         }
     }

